packages/back-end/src/controllers/reports.ts

```diff
@@ -180,6 +183,18 @@ export async function deleteReport(
     req.checkPermissions("superDelete");
   }
 
+  const connectedExperiment = await getExperimentById(
+    context,
+    report.experimentId || ""
+  );
+
+  if (!connectedExperiment) {
+    throw new Error("Could not find connected experiment");
+  }
```

Comment:
We don't want to block deleting a report if the experiment doesn't exist.  For example, if they create a report, then delete the experiment, they will be left in an orphaned state where they can't get rid of the report.

Reply:
Ah, yea - good call.

I think there could be an edge case bug here.

Ex: User's global role is "collaborator" and in Project A they have an "experimenter" role.

If they're trying to delete a report connected to an experiment in Project A - if project A is deleted, they'll get a permission error because we'll be checking the user's global role (which doesn't provide permission to delete reports.